src/Settings/ProjectSettings.ts

```diff
@@ -23,10 +24,30 @@ export class ProjectSettings extends Settings {
     }
 
     /**
-     * Gets a value by key from Azure DevOps.
+     * Gets the value of the given key from the given map of existing Azure DevOps project settings. If the key
+     * does not exist, then the given default value is returned.
      */
-    public get(key: string): PromiseLike<string> {
-        return super.get(this.getProjectSpecificKey(key));
+    public getOrDefault(key: string, existingSettings : Map<string, string>, defaultValue: string): string {
```

Comment:
issue: We still need to overwrite get here as it's still exposed via the super class. If we don't do that, it's quite easy to use `projectSettings.get` without having the project specific kep prepended which will result in weird behavior from the caller side.

This will probably be nullified if we decide to implement this logic in `Settings` (see my other comment).